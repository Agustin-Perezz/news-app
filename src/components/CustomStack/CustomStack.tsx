import React from 'react';
import Stack, { StackProps } from '@mui/material/Stack';

type CustomStackProps = {
  direction: StackProps['direction'];
  justifyContent: StackProps['justifyContent'];
  alignItems: StackProps['alignItems'];
  marginBottom?: number;
  backgroundColor?: string;
};

type Props = {
  stackProps: CustomStackProps;
  children: React.ReactNode;
};

export const CustomStack: React.FC<Props> = ({ stackProps, children }) => {
  return <Stack {...stackProps}>{children}</Stack>;
};
