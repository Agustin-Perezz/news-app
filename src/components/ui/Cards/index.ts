export * from './ListCards';
export * from './customStyleCard';
