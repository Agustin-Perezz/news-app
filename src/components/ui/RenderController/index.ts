export * from './LoadingSkeleton';
export * from './RenderController';
