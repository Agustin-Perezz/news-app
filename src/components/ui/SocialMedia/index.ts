export * from './SocialMedia';
export * from './data-media';
