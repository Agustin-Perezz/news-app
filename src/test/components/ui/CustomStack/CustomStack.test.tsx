import { render, screen } from '@testing-library/react';
import { CustomStack } from '../../../../components/ui/CustomStack';

describe('Test in <CustomStack />', () => {
  let globalContainer: HTMLElement;
  beforeEach(() => {
    const { container } = render(
      <CustomStack
        stackProps={{
          alignItems: 'center',
          direction: 'row',
          justifyContent: 'space-around',
        }}
      >
        <h2> Title example </h2>
      </CustomStack>
    );
    globalContainer = container;
  });

  it('should match snapshot', () => {
    expect(screen).toMatchSnapshot();
  });
  it('must contain past props', () => {
    const element = globalContainer.getElementsByClassName('css-1sa62bg-MuiStack-root');
    expect(element[0]).toHaveStyle('justify-content: space-around');
    expect(element[0]).toHaveStyle('align-items: center');
  });
  it('should render a children', () => {
    expect(screen.getByText('Title example')).toBeInTheDocument();
    expect(screen.getByText('Title example')).toContainHTML('h2');
  });
});
