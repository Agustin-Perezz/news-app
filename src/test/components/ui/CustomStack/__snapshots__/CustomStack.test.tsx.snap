// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CustomStack /> should match snapshot 1`] = `
<div>
  <div
    class="css-1sa62bg-MuiStack-root"
  >
    <h2>
       Title example 
    </h2>
  </div>
</div>
`;
